### ISSUE TITLE
G304: error persists after running filePath.Clean on a separate line from os.OpenFile

### ISSUE THREAD

**Issue by ericcornelissen** [securego/gosec#506]
### Summary 

Due to the recent addition of #488 I got a new error in my code. I wanted to fix this in accordance with [the docs for G304](https://securego.io/docs/rules/g304.html) but to my surprise [my solution (below) didn't work](https://github.com/ericcornelissen/wordrow/runs/906057850#step:4:58):

```go
cleanFilePath := filePath.Clean(filePath)
osFile, err := os.OpenFile(cleanFilePath, flag, mode)
```

Only after rewriting this as:

```go
osFile, err := os.OpenFile(filePath.Clean(filePath), flag, mode)
```

did [gosec recognize the solution](https://github.com/ericcornelissen/wordrow/runs/906066659).

### Steps to reproduce the behavior

Create a sample project with the file below and run gosec

```go
package main

import (
    "os"
    "path/filepath"
)

func main() {
    repoFile := "path_of_file"
    cleanRepoFile := filepath.Clean(repoFile)
    byContext, err := os.OpenFile(cleanRepoFile, os.O_RDONLY, 0600)
    if err != nil {
        panic(err)
    }
}
```

### gosec version

The gosec GitHub Action as [linked above](https://github.com/ericcornelissen/wordrow/runs/906057850).

### Go version (output of 'go version')

Same as in the gosec GitHub Action.

### Operating system / Environment

The gosec GitHub Action container.

### Expected behavior

No warning if the variable is actually the result of `filepath.Clean`.

### Actual behavior

A warning even if the variable is actually the result of `filepath.Clean`.

**Comment by benma** [comment]
I would be more interested why `filepath.Clean` would fix any security issue? It just returns an equivalent path but in a canonical form. If an attacker controlled the filename, then Clean() does not help, or am I missing something?

**Comment by ericcornelissen** [comment]
That is a fair question which I don't know the answer to (though I think you're right). This has also been discussed in #439 (in more depth). I would argue that comment deserves a separate issue, but I haven't opened such an issue as the rule is actually not relevant to my current context, but I discovered this bug in the process of realizing that :upside_down_face: (as an aside, to contribute to that discussion, the note was added to the website/docs in https://github.com/securego/securego.github.io/pull/20)

**Comment by ccojocar** [comment]
There is a bug, see the attached pull request. The variable assignment is not resolved in order to check if the `filepath.Clean` is called on the provided input.
